### PR TITLE
Adding a GitHub Action for deploying a CRIB environment

### PR DESCRIPTION
**Pull request by njegosrailic**
## What 

Adding a GitHub Action for deploying a CRIB environment.

## Why

We would like to lunch it from PR flow and spin up an environment on demand.

### CODE REVIEW

File: actions/setup-crib-environment/action.yml [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            exit 1
+            ;;
+        esac
+
+        echo "All required inputs are provided and valid."
+
+    - name: Checkout the repository
+      uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+      with:
+        repository: "smartcontractkit/crib"
```
Comment (momentmaker) [inline actions/setup-crib-environment/action.yml:84]: we will need GATI for this no?

File: actions/setup-crib-environment/action.yml [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+    - name: Setup Nix
+      uses: smartcontractkit/.github/actions/setup-nix@4df907a307d91761c15cdff65e508145bdbcfca3 # setup-nix@0.3.0
+      with:
+        cache-pubkey: ${{ inputs.cache-pubkey }}
+        enable-aws: true
+        aws-region: ${{ inputs.aws-region }}
+
+    - name: Nix Develop
+      uses: nicknovitski/nix-develop@a2060d116a50b36dfab02280af558e73ab52427d # nix-develop@1.1.0
```
Comment (momentmaker) [inline actions/setup-crib-environment/action.yml:109]: nicknovitski/nix-develop@a2060d116a50b36dfab02280af558e73ab52427d # v1.1.0

File: actions/setup-crib-environment/action.yml (hunk 1 above)
Comment (njegosrailic) [reply actions/setup-crib-environment/action.yml:84]: I think you are right.

File: actions/setup-crib-environment/action.yml (hunk 2 above)
Comment (njegosrailic) [reply actions/setup-crib-environment/action.yml]: 👍

File: actions/setup-crib-environment/action.yml [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    description: Disable namespace deletion for debugging purposes.
+    required: false
+  ecr-private-registry:
+    default: ""
+    description: ECR private registry account ID
+    required: false
+  ecr-private-registry-stage:
+    default: ""
+    description: ECR private registry account ID in Staging
+    required: false
```
Comment (scheibinger) [inline actions/setup-crib-environment/action.yml:31]: these 2 seems to be duplicated, the first one is not used in the workflow

File: actions/setup-crib-environment/action.yml [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+  steps:
+    - name: Check Required Inputs and Validate Product
+      shell: bash
+      run: |
+        # Validate required inputs
+        [[ "${{ inputs.aws-role-arn }}" ]] || { echo "aws-role-arn input is empty" ; exit 1; }
+        [[ "${{ inputs.api-gateway-host }}" ]] || { echo "api-gateway-host input is empty" ; exit 1; }
+        [[ "${{ inputs.aws-region }}" ]] || { echo "aws-region input is empty" ; exit 1; }
+        [[ "${{ inputs.ingress-base-domain }}" ]] || { echo "ingress-base-domain input is empty" ; exit 1; }
+        [[ "${{ inputs.k8s-cluster-name }}" ]] || { echo "k8s-cluster-name input is empty" ; exit 1; }
```
Comment (chainchad) [inline actions/setup-crib-environment/action.yml:66]: Is it worth validating inputs that are already required like `ingress-base-domain`? I might suggest skipping those as GHA will fail if a required input is not provided.

**Review by chainchad — COMMENTED** [review]
Looks good but echoing @momentmaker's comment about how we'll clone. If you prefer not to include GATI functionality in here, you can add as an input a token that gets provided by the calling workflow that allows cloning the /crib repo.

**Review by scheibinger — DISMISSED** [review]
Great work 🚀

File: actions/setup-crib-environment/action.yml (hunk 3 above)
Comment (njegosrailic) [reply actions/setup-crib-environment/action.yml:31]: I think it's used by the proxy image inside the GAP action, right? That's the reason I have added it.

File: actions/setup-crib-environment/action.yml (hunk 4 above)
Comment (njegosrailic) [reply actions/setup-crib-environment/action.yml:66]: Thanks. Still learning ;)

File: actions/setup-crib-environment/action.yml (hunk 1 above)
Comment (njegosrailic) [reply actions/setup-crib-environment/action.yml:81]: Add an input for Github token as Chad suggested. Please see the comment below.

File: actions/setup-crib-environment/action.yml [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+            ;;
+        esac
+
+    - name: Checkout the repository
+      uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+      env:
+        GITHUB_TOKEN: ${{ inputs.github-token }}
+      with:
+        repository: "smartcontractkit/crib"
+        ref: "main"
```
Comment (chainchad) [inline actions/setup-crib-environment/action.yml:82]:
```suggestion
      with:
        repository: "smartcontractkit/crib"
        ref: "main"
        token: ${{ inputs.github-token }}
```

**Review by chainchad — APPROVED** [review]

**Review by chainchad — APPROVED** [review]

**Pull request merged**